R/dt_spanners.R

```diff
@@ -19,21 +19,28 @@ dt_spanners_init <- function(data) {
     vars = list(),
     # The spanner label
     spanner_label = list(),
+    # The spanner id
+    spanner_id = list(),
     # Should be columns be gathered under a single spanner label?
     gather = logical(0),
     built = NA_character_
   ) %>%
     dt_spanners_set(spanners = ., data = data)
 }
 
-dt_spanners_add <- function(data, vars, spanner_label, gather) {
+dt_spanners_add <- function(data,
+                            vars,
+                            spanner_label,
+                            spanner_id,
+                            gather) {
 
-  data %>%
-    dt_spanners_get() %>%
+
+  dt_spanners_get(data = data) %>%
     dplyr::bind_rows(
       dplyr::tibble(
         vars = list(vars),
         spanner_label = list(spanner_label),
+        spanner_id = list(spanner_id),
```

Comment:
Let's just do character() for this

Reply:
This is now changed and many internal changes were made to ensure that the IDs were used instead of the labels.